routes/orders.js

```diff
@@ -0,0 +1,47 @@
+const express = require("express");
+const ordersRouter = express.Router();
+const { requireUser, isAdmin } = require("./utils");
+
+ordersRouter.get("/orders", [requireUser, isAdmin], async (req, res, next) => {
+  try {
+    const orders = await getAllOrders();
+    res.send(orders);
+  } catch (error) {
+    next(error);
+  }
+});
+
+ordersRouter.get("/orders/cart", [requireUser, isAdmin], async (req, res, next) => {
+    try {
+      const orders = await getPendingOrdersByUsers();
+      res.send(orders);
+    } catch (error) {
+      next(error);
+    }
+  }
+);
+
+ordersRouter.post("/orders", requireUser, async (req, res, next) => {
+  try {
+    const orders = await createNewOrder(productId, orderId, price, quantity);
+    res.send(orders);
+  } catch (error) {
+    next(error);
+  }
+});
+
+ordersRouter.get(
+  "/users/:userId/orders",
+  requireUser,
+  async (req, res, next) => {
+    const { userId } = req.params;
+    try {
```

Comment:
Further check to make sure the `userId` matches `req.user.id` OR `req.user.isAdmin`